src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -108,7 +109,16 @@ public void run(ActionContext context) throws Exception {
     LOG.debug("The BigQuery SQL is {}", config.getSql());
 
     // Wait for the query to complete
-    queryJob.waitFor();
+    try {
+      queryJob.waitFor();
+    } catch (BigQueryException e) {
+      // Minor bug with BigQuery java client causing it to prematurely throw an exception
+      // Pipeline should always fail if an exception is encountered, but BigQueryErrors take priority
+      // See CDAP-17061
+      if (queryJob == null || queryJob.getStatus() == null || queryJob.getStatus().getError() == null) {
```

Comment:
queryJob will never be null (otherwise a NullPointerException would have been thrown earlier) so can remove the first check.

Reply:
Fixed